sdk/ml/azure-ai-ml/azure/ai/ml/entities/_datastore/azure_storage.py

```diff
@@ -67,7 +67,7 @@ def __init__(
         endpoint: str = _get_storage_endpoint_from_metadata(),
         protocol: str = HTTPS,
         properties: Optional[Dict] = None,
-        credentials: Union[AccountKeyConfiguration, SasTokenConfiguration],
+        credentials: Optional[Union[AccountKeyConfiguration, SasTokenConfiguration]] = None,
```

Comment:
Making a required param optional is a breaking change. I can elaborate on this a bit more if you like, but we should get input from DevDiv on the best path forward here. The most you can do is expand the type hint for `credentials` to include `NoneCredentialConfiguration`

@annatisch, can you chime in here?

Reply:
I think this change should be okay. No existing customers should be broken by this - and because the constructor used keyword-only params, this doesn't introduce any awkward positional-param issues.
It's definitely breaking to change a param from optional->required, however required->optional can often be fine, especially where keyword-only params are used.